### PR TITLE
ALEX_P-week2-NodeJS

### PR DESCRIPTION
**Pull request by op117**
What was done:
- [added API key constant for openweathermap.org](https://github.com/HackYourAssignment/Node.js-Cohort50/commit/374c943cb238e202283768cd577a3a3882e3c23d)
- [separated application and server](https://github.com/HackYourAssignment/Node.js-Cohort50/commit/e3824b4cf76188e5f4612e3dcac9b09244cac6c4)
- [created test environment - jest, supertest, babel](https://github.com/HackYourAssignment/Node.js-Cohort50/commit/136f95d32cb7db2e81e56bd96a03148944787fae)
- [added tests and some fixes](https://github.com/HackYourAssignment/Node.js-Cohort50/commit/fd3e89e6b34f7154413004c82194ff775c1a91bd)
- and some other small things 
- tests have been performed, expected results correspond to actual results

### CODE REVIEW

File: assignments/hackyourtemperature/src/keys.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+const API_KEY = 'b60c6ecaef7c6de8892a9adc090288b3';
```
Comment (sarlam) [inline assignments/hackyourtemperature/src/keys.js:1]: We usually do not add the key itself to git for security reason (imagine you would pay for any call made with this key, someone could use it and you would be the one paying)

**It is not a blocker for this exercise.**

What you might want to do is either change this line to something like:
```suggestion
const API_KEY = 'INSERT_YOUR_KEY_HERE';
```

With the risk that one day someone forget and commits its change into git, or if you have extra time to look into `.env` https://www.npmjs.com/package/dotenv

File: assignments/hackyourtemperature/__tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import app from '../app.js';
+import supertest from 'supertest';
+
+const request = supertest(app);
+
+describe('POST /weather', () => {
```
Comment (sarlam) [inline assignments/hackyourtemperature/__tests__/app.test.js:6]: This looks really good :)

**Review by sarlam — COMMENTED** [review]
The nice work you put in from week 1 pay off really well here, your code look nice. 

There is two blocking things that need to be done before this can be approved:

  - The PR contains the week1 folder (and the commits that goes with it) please clean up your git branch to only have your assignment in this PR
  - There is an issue in the code that makes it fail at startup

    ```bash
    ~\Dev\Projects\HackYourAssignment\Node.js-Cohort50\assignments\hackyourtemperature> yarn start  
    yarn run v1.22.19
    $ node server.js
    file:///C:/Users/paulj/Dev/Projects/HackYourAssignment/Node.js-Cohort50/assignments/hackyourtemperature/app.js:7
    const fetch = require('node-fetch');
    ```
    
Let me know if you need any help fixing those two issues.

**Comment by op117** [comment]
Hi @sarlam ,  as for:

~\Dev\Projects\HackYourAssignment\Node.js-Cohort50\assignments\hackyourtemperature> yarn start  
yarn run v1.22.19
$ node server.js
file:///C:/Users/paulj/Dev/Projects/HackYourAssignment/Node.js-Cohort50/assignments/hackyourtemperature/app.js:7
const fetch = require('node-fetch');

image

Please, use npm install command before running the code

**Comment by op117** [comment]
As for: 
- The PR contains the week1 folder (and the commits that goes with it) please clean up your git branch to only have your assignment in this PR

Since the PR of the first week could not be merged, and I had to work with the same files, I created a new branch from the branch of the first week.
So of course you will see all the commits that make changes in different repositories.

**Comment by op117** [comment]
I see, I used "type": "module". So I changed 'required' to 'import' of node-fetch. Done.

**Pull request closed without merging**